subsys/net/openthread/rpc/client/ot_rpc_diag_client.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ot_rpc_ids.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <openthread/error.h>
+#include <openthread/dataset.h>
+#include <openthread/instance.h>
+#include <openthread/link.h>
+#include <openthread/thread.h>
+#include <sys/_stdint.h>
+
+NRF_RPC_GROUP_DECLARE(ot_group);
+
+static char version[256]; //to validate
+static otExtAddress mac = {.m8 = {0xF4, 0xCE, 0x36, 0x24, 0x95, 0x6A, 0xD1, 0xD0}};
+static otExtendedPanId ext_pan_id;
+static char thread_network_name[256];
+static otMeshLocalPrefix mesh_prefix;
+
+static void get_uint_t(ot_rpc_id id, void *result, size_t result_size) {
```

Comment:
Consider extracting these helpers into some `client/ot_rpc_common_getter.h` (and adding `ot_rpc_` prefix to the function names) so that we can reuse them in other command groups.